### PR TITLE
Add STW team perks and schematics

### PR DESCRIPTION
**Pull request by taradinoc**
Team perks and schematics from a STWProfile are exposed through the new STWTeamPerk and STWSchematic item classes. STWSchematic is further refined into subclasses for ranged weapons, melee weapons, and traps.

Enums are added for the schematic-related fields whose values don't always line up with the in-game descriptions (e.g. "blunt_hammer" weapons are called "hardware" in game), as well as for hero type (since "commando" is called "soldier" in game).

This PR also cherry-picks the fix to generateExports.js from master, and fixes the typo in "missionAlertRedemtionRecord".

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -24,6 +24,8 @@ module.exports = {
     'function-paren-newline': 'off',
     'function-call-argument-newline': 'off',
     'default-param-last': 'off',
+    'no-redeclare': 'off',
```
Comment (ThisNils) [inline .eslintrc.js:27]: Are those actually neccesary? If possible, replace them with eslint-ignore comments (or even better, fix the issue without any changes to the linting rules)

File: src/structures/STWHero.ts [hunk 2]
```diff
@@ -84,13 +84,6 @@ class STWHero extends STWItem {
     this.isRefundable = data.attributes.refundable;
   }
 
-  /**
-   * The hero's rarity value (number from 0-6).
-   */
-  public get rarityValue() {
```
Comment (ThisNils) [inline src/structures/STWHero.ts:90]: Why was this removed?

File: src/util/Util.ts [hunk 3]
```diff
@@ -360,6 +363,130 @@ export const calcSTWNonSurvivorPowerLevel = (rarity: STWItemRarity, level: numbe
...
+      break;
+    case 'pistol':
+    case 'shotgun':
+    case 'smg':
+    case 'sniper':
+      type = 'ranged';
+      subType = firstField;
+      break;
+    default:
+      // uh oh
```
Comment (ThisNils) [inline src/util/Util.ts:469]: Maybe replace that with a comment that is a bit more... helpful

**Review by ThisNils — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (taradinoc) [reply .eslintrc.js:27]: I disabled that rule to replace it with the TypeScript version of the same rule. The [ESLint documentation](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-redeclare.md) says "note you must disable the base rule as it can report incorrect errors".

Specifically, the base `no-redeclare` rule reports these errors:

```
...\fnbr.js\src\structures\STWMeleeWeaponSchematic.ts
  9:7  error  'STWMeleeWeaponSchematic' is already defined  no-redeclare

...\fnbr.js\src\structures\STWRangedWeaponSchematic.ts
  9:7  error  'STWRangedWeaponSchematic' is already defined  no-redeclare

...\fnbr.js\src\structures\STWTrapSchematic.ts
  10:7  error  'STWTrapSchematic' is already defined  no-redeclare

D:\working\github\fnbr.js\src\structures\STWWeaponSchematic.ts
  10:7  error  'STWWeaponSchematic' is already defined  no-redeclare

✖ 4 problems (4 errors, 0 warnings)
```

Those files define an interface and a class with the same name; the interface constrains the types of fields that were introduced in the base class. It's a valid pattern in TypeScript, and I don't know of a better way to write it, but ESLint's base rules are unaware of TypeScript declaration merging.

File: src/util/Util.ts (hunk 3 above)
Comment (taradinoc) [reply src/util/Util.ts]: 👍

File: .eslintrc.js (hunk 1 above)
Comment (ThisNils) [reply .eslintrc.js:27]: I think it's not necessary to use an interface. Take a look at <https://github.com/fnbrjs/fnbr.js/blob/master/src/structures/SentFriendMessage.ts>.
In my opinion that's also a less complex way than defining an interface that provides the overwritten types

**Review by ThisNils — APPROVED** [review]
